Feature/components

Kazuma-47 opened the pull request:
[WIP] 
the scripts i want to check:
SequenceManager
SequenceItem
ColorSequenceManager
SequenceButton
EnumColor

Kazuma-47 committed:
added button interface

RossTubes committed:
pressurePlate

simple script and scene made for the pressure plate test feature.

Kazuma-47 committed:
added TMP and dynamic sequence scripts

Kazuma-47 committed:
updated sequence puzzle and prapared it for merging

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,117 @@
+            if (Sequence[i] != EnumColor.Unassigned) continue;
+
+            EnumColor color;
+            color = GetRandomColor();
```
```suggestion
           EnumColor color = GetRandomColor();
```
Creating en Assignment samen, dit is meer preference maar ik denk tenzij je vooraf iets moet checken zoals null check of string.empty checken bij string dan kan je meestal gwn samen doen

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,117 @@
+    public bool CheckFinalSequenceInput(int currentLength)
+    {
+        if (currentLength + 1 == sequenceLength) return true;
+        else return false;
```
```suggestion
        return (currentLength + 1 == sequenceLength)
```
dit kan toch?

Kazuma-47 committed:
forgot to save scene

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,117 @@
+        }
+        if (isCorrect)
+        {
+            if (sequenceLength == Sequence.Length) onSolved?.Invoke();
+            else NextRound();
```
```suggestion
            if (sequenceLength == Sequence.Length) 
            {
                onSolved?.Invoke();
            }
           else 
           { 
              NextRound();
           }
```

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/SequenceButton.cs:
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+public class SequenceButton : SequenceItem<EnumColor> 
```
```suggestion
public sealed class SequenceButton : SequenceItem<EnumColor> 
```

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/SequenceButton.cs:
```diff
@@ -0,0 +1,12 @@
+
+public class SequenceButton : SequenceItem<EnumColor> 
+{
+    public EnumColor ButtonValue;
```
```suggestion
    [SerializeField] private EnumColor value;
```
kan private of nee ik zie nergens in andere scripts

DanielNijkamp commented on Assets/Scripts/Framework/puzzels/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,117 @@
+
+public sealed class ColorSequenceManager : MonoBehaviour, SequenceManager<EnumColor>
+{
+    public EnumColor[] Sequence { get; set; }  = new EnumColor[5];
+    public EnumColor[] SubmittedSequence { get; set; } = new EnumColor[5];
```
```suggestion
    public EnumColor[] Sequence { get; set; }  = new EnumColor[5];
    public EnumColor[] SubmittedSequence { get; set; } = new EnumColor[5];
```
moeten deze public of kunnen deze ook private (serialized)worden?

Kazuma-47 replied:
deze kunnen niet private omdat het komt van de interface

DanielNijkamp reviewed (commented):
@Kazuma-47 Paar comments van jouw scripts, verder ziet wel gwn super uit

RossTubes committed:
Timed pressure plate puzzel

I fixed the timer sequence for the pressure plate puzzle if you don't activate the pressure plate before the timer ends then the game over function will activate.

Kazuma-47 committed:
code review feedback verwerkt

RossTubes committed:
Lights Puzzles

first version of the lights puzzle to conncect all the objects with eachoter

RossTubes committed:
Light puzzle Test version

worked on the light connection puzzle it does not work perfectly yet.

DanielNijkamp committed:
Changed lights puzzle to toggelable

DanielNijkamp committed:
Changed pressurePlate into TriggerCollider

Eliza-van-bokhoven reviewed (commented):
Seems pretty solid, @Kazuma-47, @DanielNijkamp 
![thumbs up](https://github.com/DanielNijkamp/PVB/assets/83960112/1557a2f0-c546-4cc7-b92d-45caddbce09d)

Kazuma-47 committed:
finished symbol light puzzle

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/LightSequencePuzzle/SequenceTile.cs:
Deze mag weg neem ik aan?

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,120 @@
+    [SerializeField] private UnityEvent onSolved = new UnityEvent();
+    [SerializeField] private UnityEvent onReset = new UnityEvent();
+
+    private int sequenceLength = 3;
```
```suggestion
    [SerializeField] private int sequenceLength;
```
Beter om instelbaar te maken via inspector

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,120 @@
+                color = GetRandomElement();
+            }
+            Sequence[i] = color;
+            break;
```
```suggestion
```
Ik denk niet dat deze break hierin hoort of uberhaupt nodig is.

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,120 @@
+                SubmittedSequence[i] = value;
+                if (CheckFinalSequenceInput(i))
+                        CheckSubmittedSequence();
+                break;
```
```suggestion
```
Kan hier niet gereturned worden ipv break?

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs:
```diff
@@ -0,0 +1,120 @@
+                return true;
+            }
+        }
+        return false;
```
```suggestion
        return Sequence.Any(c => c == color)
```

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SequencePuzzle/ColorSequenceManager.cs:
Check of je paar methods misschien private kan maken anders dan is het niet zo erg als dat niet mogenlijk is

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SymbolPuzzle/VisibilitySymbol.cs:
Ik snap als deze script word gebruikt om te testen, ik zou dit voor nu weghalen maar misschien dat we deze functionaliteit later kunnen terug brengen door middel van UI modals voor de interactable of iets anders

DanielNijkamp commented on Assets/Scripts/Framework/Puzzles/SymbolPuzzle/SymbolTile.cs:
```diff
@@ -0,0 +1,21 @@
+public class SymbolTile : SequenceItem<EnumSymbols>
+{
+    [SerializeField] private EnumSymbols itemValue;
+    [HideInInspector] public bool isActive = false;
```
```suggestion
    [HideInInspector] public bool isActive;
```

DanielNijkamp reviewed (changes requested):
Paar comments maar wel goeie code. nog op maandag 15/04/2024 checken of alles werkt binnen unity

Kazuma-47 committed:
applied code review feedback

Kazuma-47 committed:
cleaned some code with name convention

Kazuma-47 committed:
removed empty script

Kazuma-47 committed:
sealed classes

DanielNijkamp reviewed (approved):
I very like habibi
